pkg/trace/otel.go

```diff
@@ -104,16 +104,50 @@ func (t *otelTracer) initTracer(ctx context.Context, serviceName string) error {
 		log.Error(ctx, "Unable to configure trace provider", events.NewErrorInfo(err))
 	}
 
-	tp := sdktrace.NewTracerProvider(
-		sdktrace.WithBatcher(exp),
-		sdktrace.WithResource(r),
-		// accepts sample rates as number of requests seen per request sampled
-		sdktrace.WithSampler(forceSample(uint(100/t.Otel.SamplePercent))),
-		sdktrace.WithSpanProcessor(Annotator{
-			globalTags: t.GlobalTags,
-			sampleRate: int64(100 / t.Otel.SamplePercent),
-		}),
-	)
+	var tp *sdktrace.TracerProvider
+	if t.Otel.AdditionalEndpoint != "" {
+		key, err := t.Otel.AdditionalAPIKey.Data(ctx)
+		if err != nil {
+			log.Error(ctx, "Unable to fetch additional otel API key", events.NewErrorInfo(err))
+		}
+
+		headers := map[string]string{
+			"lightstep-access-token": strings.TrimSpace(string(key)),
+		}
+
+		client := otlptracegrpc.NewClient(
+			otlptracegrpc.WithEndpoint(t.Otel.AdditionalEndpoint),
+			otlptracegrpc.WithHeaders(headers),
+		)
+
+		aexp, err := otlptrace.New(ctx, client)
+		if err != nil {
+			log.Error(ctx, "Unable to start additional trace exporter", events.NewErrorInfo(err))
+		}
+
+		tp = sdktrace.NewTracerProvider(
+			sdktrace.WithBatcher(exp),
+			sdktrace.WithBatcher(aexp),
+			sdktrace.WithResource(r),
+			// accepts sample rates as number of requests seen per request sampled
+			sdktrace.WithSampler(forceSample(uint(100/t.Otel.SamplePercent))),
+			sdktrace.WithSpanProcessor(Annotator{
+				globalTags: t.GlobalTags,
+				sampleRate: int64(100 / t.Otel.SamplePercent),
+			}),
+		)
+	} else {
+		tp = sdktrace.NewTracerProvider(
```

Comment:
Could we combine this and L128 by making the options a slice that we add to instead? I don't really like duplicating the code here :'(

Reply:
That's totally fair. Will do.